calcom/cal.com
# feat: ApplePay as Payment Option for Paid Bookings

## What does this PR do?


This PR adds the ability to accept apple pay as payment option for paid bookings.

- Fixes https://github.com/calcom/cal.com/issues/15276
- Fixes [CAL-3839]





## Mandatory Tasks (DO NOT REMOVE)
Apple pay             |  Apple pay
:-------------------------:|:-------------------------:
![Screenshot 2024-06-02 at 11 37 37 PM](https://github.com/calcom/cal.com/assets/70239736/d05b20d4-5208-485a-aa3e-f367a16099b0)  |  ![Screenshot 2024-06-02 at 11 38 40 PM](https://github.com/calcom/cal.com/assets/70239736/a3db6ad6-8b27-43d5-88d3-27bc2dc7eb34)


## How should this be tested?



- Apple Pay only runs on https so you'll need a way to run your app through https to test locally. You can use ngrok for this and what you need to do is go to your [payment method domains](https://dashboard.stripe.com/settings/payment_method_domains?enabled=true) and add a domain if you don't have one already, once you add your domain you'll need to put the file that stripe gives you under `.well-known` so you'll create a folder in `apps/web/public/.well-known` and paste the file that stripe gives you there, once you do that you should be good to go. If you are gonna test on desktop, keep in mind that apple pay only works on safari, this `PaymentRequestButtonElement` also works we the following providers (image below) so it's up to the company to decide which to accept but all of this are set up through the dashboard so you could add as many as you want.
![Screenshot 2024-06-02 at 11 48 32 PM](https://github.com/calcom/cal.com/assets/70239736/c48a51a8-2b6e-499a-883a-e9cf7736ebd1)
